jdaviz/core/tools.py

```diff
@@ -28,6 +31,26 @@
 BqplotYRangeMode.icon = os.path.join(ICON_DIR, 'select_y.svg')
 
 
+class InteractCheckableToolMultiViewer(InteractCheckableTool):
+    def activate(self):
+        # skip the default behavior in InteractCheckableTool that disables
+        # any active tool in other viewers
+        self.viewer._mouse_interact.next = self.interact
+
+
+# override InteractCheckableTool so everything that inherits from it will get our
+# custom version instead
+tools.InteractCheckableTool = InteractCheckableToolMultiViewer
+
+# but the classes have already been created that inherit from the original class,
+# so we need to override those as well (BqplotEllipseMode inherits from BqplotCircleMode
+# so doesn't need this)
+for cls in [BqplotCircleMode, BqplotPanZoomMode, BqplotPanZoomXMode, BqplotPanZoomYMode,
+            BqplotRectangleMode, BqplotCircleMode, BqplotXRangeMode,
+            BqplotYRangeMode, BqplotSelectionTool, ROIClickAndDrag, HomeTool]:
+    cls.__bases__ = (InteractCheckableToolMultiViewer,)
+
+
```

Comment:
on a scale from 1-10, how hideously hacky is this?  And on the same scale from 1-10, how much hackiness are we okay with? 😂 

Previously, the nested toolbar interacted with the unused - but still instantiated - `BasicJupyterToolbar` toolbar, and so somewhat "accidentally" resulted in tools remaining active when activating a tool in another viewer.  I think this is actually the behavior that we want, especially with things like the slice tool being a persistent "default".  However, with the original toolbar now removed entirely, we need a different implementation to override the glue-jupyter default.  If this is too ugly, our options are:
* change the behavior to adopt glue's behavior.  We would then likely need to change the slice tool from a tool in the toolbar to the default mode of that viewer (replace `ROIClickAndDrag`), but then lose the ability to disable that tool and interact with subsets.  I personally am not in favor of this option unless/until we migrate to a global toolbar.
* propose changes upstream to glue-jupyter, perhaps a switch somewhere that toggles whether tools should deactivate tools in other viewers
* something else that I'm not considering

Reply:
I am afraid if we do it this way, and then a new dev comes and add a new tool but forgot to do this part, then things will  break.

How about we see if @astrofrog has any ideas?

Reply:
All for ideas from astrofrog (I was definitely going to bring it up on Tuesday).  I'll just add that there shouldn't be concern of a tool implemented in jdaviz breaking, since it will automatically inherit correctly.  But a new tool implemented in glue-jupyter and then used or inherited within jdaviz could definitely cause confusion if it isn't added to this list.  Definitely a valid concern.